mantidproject/mantid
# Groundwork for visualising predicted peaks

This is a sub issue of #15435.

This adds the groundwork for visualising predicted peaks which fall outside of an instrument's detector geometry.

**To test:**

 - Check key tests pass
   - `PredictPeaks`
   - `PeakTest`

 - You can enable the new feature by adding an entry to an IDF similar to the following example for WISH:

```xml
   
   <component type="extended-detector-space">
       <location /> 
   </component>

   <type name="extended-detector-space">
       <properties />
       <infinite-cylinder id="detector-surface">
           <centre x="0.0" y="0.0" z="0.0" />
           <axis x="0.0" y="1.0" z="0.0" /> 
           <radius val="2.2018" />
       </infinite-cylinder> 
       <algebra val="detector-surface" />
   </type>
```
 - Check that adding that definition and running `PredictPeaks` with the new option `PredictPeaksOutsideDetectors` will allow the prediction of peaks that fall outside of defined detectors and that these are in the correct place (relative to the component defined above).

 - These peaks should also show up in an unwrapped view on the instrument view, but proper visualisation has not yet been implemented and will be handled outside the scope of this ticket (see #18923).

 - Code review

Fixes #18920.

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
